### DISCUSSION TITLE
Indexing properties and case insensitive equality

### DISCUSSION THREAD

**Discussion by peppy** [realm/realm-dotnet#2733] [Q&A] [answered]
I've searched through documentation surrounding the use of `[Indexed]` and can't find a solid answer on whether it should help for a case insensitive filter like:

```csharp
    public class RealmNamedFileUsage : EmbeddedObject, INamedFile, INamedFileUsage
    {
        [Indexed]
        public string Filename { get; set; } = null!;

        ...
    }

    public class SkinInfo : RealmObject, IHasRealmFiles, IEquatable<SkinInfo>, IHasGuidPrimaryKey, ISoftDelete, IHasNamedFiles
    {
        public IList<RealmNamedFileUsage> Files { get; } = null!;
    }

    ...
    SkinInfo skinInfo = // already retrieved managed realm object instance

    // usage
    skinInfo.Files.AsRealmQueryable().Filter(@"Filename ==[c] $0", filename).FirstOrDefault();
    // or (should be functionally the same operation)
    skinInfo.Files.AsRealmQueryable().FirstOrDefault(f => f.Filename.Equals(filename, StringComparison.OrdinalIgnoreCase))?.File.GetStoragePath();
```

Also curious if there's some kind of diagnostics which would help profile such cases. Currently I am looking at the time it takes for the `Realms.CollectionHandleBase.GetFilteredResults(String, RealmValue[])` call - does that sound correct?

**Comment by papafe** [comment, marked as answer]
Hi @peppy, unfortunately in this case the index does not have any effect. The index has an effect only when the query is defined directly on the class that contains the index (`RealmNamedFileUsage` in this case) , and not through links.

**Reply by peppy** [reply]
So in this case if we were looking for efficient lookups we'd need to switch from `EmbeddedObject` to `RealmObject` and do a top level query on the realm context itself?

Can you confirm that the index works for all `StringComparison` modes in such a case?

**Reply by papafe** [reply]
Yes, you're correct. And yes, the index should work for both case sensitive and case insensitive queries.
